### PR TITLE
Add support for run from blob

### PR DESCRIPTION
**Pull request by Flanker32**
Support run from blob, which is part of run from package:

plugin will upload function artifact to function storage and update the app setting, set `WEBSITE_RUN_FROM_PACKAGE = BlobURL`. This is recommanded for linux consumption and premium plan.

### CODE REVIEW

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/RunFromBlobArtifactHandlerImpl.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        super(builder);
+    }
+
+    @Override
+    public void publish(DeployTarget deployTarget) throws Exception {
+        final File zipPackage = FunctionArtifactHelper.createFunctionArtifact(stagingDirectoryPath);
+        final CloudStorageAccount storageAccount = FunctionArtifactHelper.getCloudStorageAccount(deployTarget);
+        final CloudBlockBlob blob = AzureStorageHelper.uploadFileAsBlob(zipPackage, storageAccount,
+                DEPLOYMENT_PACKAGE_CONTAINER, zipPackage.getName());
+        final String sasToken = AzureStorageHelper.getSASToken(blob, Period.ofYears(10));
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/RunFromBlobArtifactHandlerImpl.java]: make 10 as a const

Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/RunFromBlobArtifactHandlerImpl.java]: Done

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java [hunk 2]
```diff
@@ -40,6 +43,15 @@ public static File createFunctionArtifact(final String stagingDirectoryPath) thr
         return zipPackage;
     }
 
+    public static void updateAppSetting(final DeployTarget target, final String key, final String value) throws MojoExecutionException {
+        final WebAppBase webAppBase = target.getApp();
+        if (!(webAppBase instanceof FunctionApp)) {
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: why webApp instance of Function the name is not proper

File: azure-functions-maven-plugin/src/test/java/com/microsoft/azure/maven/function/DeployMojoTest.java [hunk 3]
```diff
@@ -240,16 +240,16 @@ public void getMSDeployArtifactHandler() throws MojoExecutionException {
...
-        doReturn("azure-functions-maven-plugin").when(mojoSpy).getPluginName();
-        doReturn("test-path").when(mojoSpy).getBuildDirectoryAbsolutePath();
-        doReturn(DeploymentType.FTP).when(mojoSpy).getDeploymentType();
-        final ArtifactHandler handler = mojoSpy.getArtifactHandler();
-
-        assertNotNull(handler);
-        assertTrue(handler instanceof FTPArtifactHandlerImpl);
-    }
+//    @Test
+//    public void getFTPArtifactHandler() throws MojoExecutionException {
```
Comment (andxu) [inline azure-functions-maven-plugin/src/test/java/com/microsoft/azure/maven/function/DeployMojoTest.java]: why remove this case?

Comment (Flanker32) [reply azure-functions-maven-plugin/src/test/java/com/microsoft/azure/maven/function/DeployMojoTest.java]: Commit by mistake, fixed in later commit.

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java (hunk 2 above)
Comment (Flanker32) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/artifact/FunctionArtifactHelper.java]: Renamed

**Pull request merged**